build_tools/cmake/build_tracing.sh

```diff
@@ -15,20 +15,28 @@ set -x
 BUILD_DIR="${1:-${IREE_TRACING_BUILD_DIR:-build-tracing}}"
 
 source build_tools/cmake/setup_build.sh
-source build_tools/cmake/setup_ccache.sh
+# Note: not using ccache since the runtime build should be fast already.
 
-# Note: https://github.com/openxla/iree/issues/6404 prevents us from building
-# tests with these other settings. Many tests invoke the compiler tools with
-# MLIR threading enabled, which crashes with compiler tracing enabled.
+echo "::group::Building with -DIREE_TRACING_PROVIDER=tracy"
 "${CMAKE_BIN?}" -B "${BUILD_DIR}" \
   -G Ninja . \
   -DPython3_EXECUTABLE="${IREE_PYTHON3_EXECUTABLE}" \
   -DPYTHON_EXECUTABLE="${IREE_PYTHON3_EXECUTABLE}" \
   -DCMAKE_BUILD_TYPE=RelWithDebInfo \
   -DIREE_ENABLE_RUNTIME_TRACING=ON \
+  -DIREE_TRACING_PROVIDER=tracy \
   -DIREE_BUILD_COMPILER=OFF
 "${CMAKE_BIN?}" --build "${BUILD_DIR}" -- -k 0
+echo "::endgroup::"
 
-if (( IREE_USE_CCACHE == 1 )); then
-  ccache --show-stats
-fi
+echo "::group::Building with -DIREE_TRACING_PROVIDER=console"
+"${CMAKE_BIN?}" -B "${BUILD_DIR}" \
+  -G Ninja . \
+  -DPython3_EXECUTABLE="${IREE_PYTHON3_EXECUTABLE}" \
+  -DPYTHON_EXECUTABLE="${IREE_PYTHON3_EXECUTABLE}" \
+  -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+  -DIREE_ENABLE_RUNTIME_TRACING=ON \
+  -DIREE_TRACING_PROVIDER=console \
```

Comment:
Seeing failures here that I can't reproduce on Windows :/

https://github.com/openxla/iree/actions/runs/7963638307/job/21739648223?pr=16454#step:4:1230

undefined reference to `iree_time_now`:
```
  FAILED: tools/iree-cpuinfo 
  : && /usr/bin/clang-9 -O2 -g -DNDEBUG -lm tools/CMakeFiles/iree-cpuinfo.dir/iree-cpuinfo.c.o -o tools/iree-cpuinfo  runtime/src/iree/base/libiree_base_base.a  runtime/src/iree/base/internal/libiree_base_internal_cpu.a  runtime/src/iree/base/libiree_base_base.a  runtime/src/iree/base/tracing/libiree_base_tracing_provider.a  -pthread && :
  /usr/bin/ld: runtime/src/iree/base/tracing/libiree_base_tracing_provider.a(console.c.o): in function `iree_tracing_zone_begin':
  /work/runtime/src/iree/base/tracing/console.c:145: undefined reference to `iree_time_now'
  /usr/bin/ld: /work/runtime/src/iree/base/tracing/console.c:145: undefined reference to `iree_time_now'
  /usr/bin/ld: runtime/src/iree/base/tracing/libiree_base_tracing_provider.a(console.c.o): in function `iree_tracing_zone_end':
  /work/runtime/src/iree/base/tracing/console.c:178: undefined reference to `iree_time_now'
  clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Unused variable:
```
  FAILED: runtime/src/iree/hal/drivers/cuda/CMakeFiles/iree_hal_drivers_cuda_cuda.objects.dir/nccl_channel.c.o 
  /usr/bin/clang-9 -DIREE_TRACING_MODE=2 -DIREE_TRACING_PROVIDER_H=\"iree/base/tracing/console.h\" -I/work -I/work/build-tracing -I/work/runtime/src -I/work/build-tracing/runtime/src -I/usr/local/iree_cuda_deps/include -I/work/third_party/nccl -I/work/build-tracing/runtime/src/iree/base/internal/flatcc -I/work/build-tracing/runtime/src/iree/schemas -isystem /work/third_party/flatcc/include -O2 -g -DNDEBUG -fPIC -fvisibility=hidden -Werror -Wall -Wno-error=deprecated-declarations -Wno-ambiguous-member-template -Wno-char-subscripts -Wno-extern-c-compat -Wno-gnu-alignof-expression -Wno-gnu-variable-sized-type-not-at-end -Wno-ignored-optimization-argument -Wno-invalid-offsetof -Wno-invalid-source-encoding -Wno-mismatched-tags -Wno-pointer-sign -Wno-reserved-user-defined-literal -Wno-return-type-c-linkage -Wno-self-assign-overloaded -Wno-sign-compare -Wno-signed-unsigned-wchar -Wno-strict-overflow -Wno-trigraphs -Wno-unknown-pragmas -Wno-unknown-warning-option -Wno-unused-command-line-argument -Wno-unused-const-variable -Wno-unused-function -Wno-unused-local-typedef -Wno-unused-private-field -Wno-user-defined-warnings -Wno-missing-braces -Wc++20-extensions -Wctad-maybe-unsupported -Wfloat-overflow-conversion -Wfloat-zero-conversion -Wfor-loop-analysis -Wformat-security -Wgnu-redeclared-enum -Wimplicit-fallthrough -Winfinite-recursion -Wliteral-conversion -Wnon-virtual-dtor -Woverloaded-virtual -Wpointer-arith -Wself-assign -Wstring-conversion -Wtautological-overlap-compare -Wthread-safety -Wthread-safety-beta -Wunused-comparison -Wvla -fno-omit-frame-pointer -pthread -I/work/third_party/flatcc/include/ -I/work/third_party/flatcc/include/flatcc/reflection/ -std=gnu11 -MD -MT runtime/src/iree/hal/drivers/cuda/CMakeFiles/iree_hal_drivers_cuda_cuda.objects.dir/nccl_channel.c.o -MF runtime/src/iree/hal/drivers/cuda/CMakeFiles/iree_hal_drivers_cuda_cuda.objects.dir/nccl_channel.c.o.d -o runtime/src/iree/hal/drivers/cuda/CMakeFiles/iree_hal_drivers_cuda_cuda.objects.dir/nccl_channel.c.o -c /work/runtime/src/iree/hal/drivers/cuda/nccl_channel.c
  /work/runtime/src/iree/hal/drivers/cuda/nccl_channel.c:558:26: error: unused variable 'collective_str' [-Werror,-Wunused-variable]
        iree_string_view_t collective_str =
                           ^
```

Reply:
Fixed the unused variable warning. Not sure how to make progress on the reference to `iree_time_now`. The code looks correct to me. Might need to set up a Linux machine to be able to reproduce and iterate on it locally.

Reply:
Tracked that down on Linux to using the system linker instead of LLD (via `-DIREE_ENABLE_LLD=ON`). Can fix that separately or as part of this 🤔